agent/agent.go

```diff
@@ -1187,27 +1187,29 @@ func (a *Agent) shouldKill(tc *taskContext, ignoreTaskGroupCheck bool) bool {
 
 // logPanic logs a panic to the task log and returns the panic error, along with
 // the original error (if any). If there was no panic error, this is a no-op.
-func (a *Agent) logPanic(logger client.LoggerProducer, pErr, originalErr error, op string) error {
+func (a *Agent) logPanic(tc *taskContext, pErr, originalErr error, op string) error {
 	if pErr == nil {
 		return nil
 	}
 
 	catcher := grip.NewBasicCatcher()
 	catcher.Add(originalErr)
 	catcher.Add(pErr)
-	if logger != nil && !logger.Closed() {
-		logMsg := message.Fields{
-			"message":   "programmatic error: Evergreen agent hit panic",
-			"operation": op,
-		}
+	logMsg := message.Fields{
+		"message":   "programmatic error: Evergreen agent hit panic",
+		"operation": op,
+	}
+	if tc.logger != nil && !tc.logger.Closed() {
 		if a.opts.HostID != "" {
 			logMsg["host_id"] = a.opts.HostID
 		}
 		if a.opts.PodID != "" {
 			logMsg["pod_id"] = a.opts.PodID
 		}
-		logger.Task().Error(logMsg)
+		tc.logger.Task().Error(logMsg)
 	}
+	logMsg["task_id"] = tc.task.ID
+	grip.Alert(message.WrapError(errors.WithStack(pErr), logMsg))
```

Comment:
Can we file a ticket to reduce the extreme spam this is likely to cause? While this will work and will give us some much-needed insight into agent panics, I suspect it may start causing ops alert to explode with panic messages in case of a bad deploy because of how often/quickly the agent runs, especially given that an agent hitting a panic is likely to keep hitting that panic over and over (multiplied by 10k+ hosts). Unfortunately, panics on the app server and agent go straight to Slack with this method, so there's no way to cut down on the spam once it's started panicking.

We could fix the spamming issue by removing the special logic that sends alert+ logs straight to Slack, and instead creating an alert in Splunk that will inform us of panics (since Splunk alerts can be tuned to determine how often we frequently we want to be re-notified).